source/docs/software/basic-programming/java-units.rst

```diff
@@ -1,58 +1,135 @@
++--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------+
+| ``Measure<Temperature>``                               | ``Kelvin``, ``Celsius``, ``Fahrenheit``                                                                                |
++--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------+
+| ``Measure<Per<Voltage, Velocity<Distance>>>``          | ``VoltsPerMeterPerSecond``                                                                                             |
++--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------+
+| ``Measure<Per<Voltage, Velocity<Velocity<Distance>>>>``| ``VoltsPerMeterPerSecondSquared``                                                                                      |
++--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------+
+| ``Measure<Per<Voltage, Velocity<Angle>>>``             | ``VoltsPerRadianPerSecond``                                                                                            |
++--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------+
+| ``Measure<Per<Voltage, Velocity<Velocity<Angle>>>>``   | ``VoltsPerRadianPerSecondSquared``                                                                                     |
++--------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------+
+
+Java Generics
+^^^^^^^^^^^^^
+Units of measurement can be complex expressions involving various dimension, such as distance, time, and velocity. Nested `generic type parameters <https://docs.oracle.com/javase/tutorial/java/generics/index.html>`__ allow for the definition of units that can represent such complex expressions. Generics are used to keep the library concise, reusable, and extensible, but it tends to be verbose due to the syntax for Java generics.
+
+For instance, consider the type ``Measure<Velocity<Distance>>``. This type represents a measurement for velocity, where the velocity itself is expressed as a unit of distance per unit of time. This nested structure allows for the representation of units like meters per second or feet per minute. Similarly, the type ``Measure<Per<Voltage, Velocity<Distance>>>`` represents a measurement for a ratio of voltage to velocity. This type is useful for representing quantities like volts per meter per second.
```

Comment:
Should mention what V/m/s would be useful for so readers know it's not contrived just for sake of an example. This would also be a good place to make the distinction between `Velocity` and `Per<, Time>`